core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -90,54 +118,60 @@ private DataTask task(StaticTableScan scan) {
 
   private static StaticDataTask.Row convertPartition(Partition partition) {
     return StaticDataTask.Row.of(
-        partition.key, partition.specId, partition.dataRecordCount, partition.dataFileCount);
+        partition.key,
+        partition.specId,
+        partition.dataRecordCount,
+        partition.dataFileCount,
+        partition.posDeleteRecordCount,
+        partition.posDeleteFileCount,
+        partition.eqDeleteRecordCount,
+        partition.eqDeleteFileCount);
   }
 
   private static Iterable<Partition> partitions(Table table, StaticTableScan scan) {
     Types.StructType normalizedPartitionType = Partitioning.partitionType(table);
     PartitionMap partitions = new PartitionMap();
 
-    // cache a position map needed by each partition spec to normalize partitions to final schema
-    Map<Integer, int[]> normalizedPositionsBySpec =
-        Maps.newHashMapWithExpectedSize(table.specs().size());
+    try (CloseableIterable<DataFile> datafiles = planDataFiles(scan);
```

Comment:
I would say, to try this if we can, both to clean the code and also as we don't do another scan. 

I think it may work by doing 'CloseableIterables.transform()' in the worst case to cast the type

Reply:
@ajantha-bhat can we see about this?  Will it work to wrap the manifest iterable to get the right type in planDataFiles()?

```
CloseableIterable.transform(ManifestFiles.read(manifest, table.io(), table.specs())
        .caseSensitive(scan.isCaseSensitive())
        .select(BaseScan.SCAN_COLUMNS).entries(), t -> (ContentFile<?>) t);
```

Reply:
updated